test/unittest/dds/xtypes/type_representation/TypeObjectUtilsTests.cpp

```diff
@@ -0,0 +1,2008 @@
+                complete_header, 256, test_identifier), InvalidArgumentError);
+    PlainCollectionHeader minimal_header = TypeObjectUtils::build_plain_collection_header(
+        EquivalenceKindValue::MINIMAL, flags);
+    // TypeIdentifier inconsistent with minimal header
+    EXPECT_THROW(PlainSequenceLElemDefn plain_seq = TypeObjectUtils::build_plain_sequence_l_elem_defn(
+                minimal_header, 256, test_identifier), InvalidArgumentError);
+    PlainCollectionHeader fully_descriptive_header = TypeObjectUtils::build_plain_collection_header(
+        EquivalenceKindValue::BOTH, flags);
+    // TypeIdentifier consistent with fully-descriptive header
+    EXPECT_NO_THROW(PlainSequenceLElemDefn plain_seq = TypeObjectUtils::build_plain_sequence_l_elem_defn(
+                fully_descriptive_header, 256, test_identifier));
+    // Change discriminator to EK_COMPLETE
+    EquivalenceHash hash;
+    test_identifier->equivalence_hash(hash);
+    // TypeIdentifier consistent with complete header
+    EXPECT_NO_THROW(PlainSequenceLElemDefn plain_seq = TypeObjectUtils::build_plain_sequence_l_elem_defn(
+                complete_header, 256, test_identifier));
+    // TypeIdentifier inconsistent with minimal header
+    EXPECT_THROW(PlainSequenceLElemDefn plain_seq = TypeObjectUtils::build_plain_sequence_l_elem_defn(
+                minimal_header, 256, test_identifier), InvalidArgumentError);
+    // TypeIdentifier inconsistent with fully-descriptive header
+    EXPECT_THROW(PlainSequenceLElemDefn plain_seq = TypeObjectUtils::build_plain_sequence_l_elem_defn(
+                fully_descriptive_header, 256, test_identifier), InvalidArgumentError);
+    // Change discriminator to EK_MINIMAL
+    EXPECT_NO_THROW(test_identifier->_d(EK_MINIMAL));
+    // TypeIdentifier inconsistent with complete header
+    EXPECT_THROW(PlainSequenceLElemDefn plain_seq = TypeObjectUtils::build_plain_sequence_l_elem_defn(
+                complete_header, 256, test_identifier), InvalidArgumentError);
+    // TypeIdentifier consistent with minimal header
+    EXPECT_NO_THROW(PlainSequenceLElemDefn plain_seq = TypeObjectUtils::build_plain_sequence_l_elem_defn(
+                minimal_header, 256, test_identifier));
+    // TypeIdentifier inconsistent with fully-descriptive header
+    EXPECT_THROW(PlainSequenceLElemDefn plain_seq = TypeObjectUtils::build_plain_sequence_l_elem_defn(
+                fully_descriptive_header, 256, test_identifier), InvalidArgumentError);
+}
+
+// Build PlainArraySElemDefn with inconsistent parameters.
+TEST(TypeObjectUtilsTests, build_plain_array_s_elem_defn_inconsistencies)
+{
+    eprosima::fastcdr::external<TypeIdentifier> test_identifier{new TypeIdentifier()};
+    CollectionElementFlag flags = TypeObjectUtils::build_collection_element_flag(TryConstructKind::TRIM, false);
+    SBoundSeq bound_seq;
+    TypeObjectUtils::add_array_dimension(bound_seq, 10);
```

Comment:
I could be insteresting to add tests in debug mode for the add___ functions. They do quite a few checks, and there is no other way to check that.

Reply:
I have included this suggestion as a nice to have, but it is not going to be implemented in this PR.